Documentation improvements for 1.0 release

FrankRay78 opened the pull request:
**The following PR https://github.com/spectreconsole/examples/pull/1 should be reviewed/merged at the same time as this one.**

---

fixes #1602

<!-- formalities. These are not optional. -->

- [X] I have read the [Contribution Guidelines](https://github.com/spectreconsole/spectre.console/blob/main/CONTRIBUTING.md)
- [X] I have commented on the issue above and discussed the intended changes
- [ ] A maintainer has signed off on the changes and the issue was assigned to me
- [X] All newly added code is adequately covered by tests
- [X] All existing tests are still running without errors
- [X] The documentation was modified to reflect the changes ~~_OR_ no documentation changes are required.~~

## Changes

- Fix broken documentation build introduced by Razor change (See: https://github.com/dotnet/razor/issues/10568 for breaking Razor change)
- Fixed broken links to the examples project from preparation for 1.0 (See: https://github.com/spectreconsole/spectre.console/issues/1602#issuecomment-2276622800)
- Rewrote the CommandApp Dependency Injection section for better clarity and readability.

---
Please upvote :+1: this pull request if you are interested in it.

patriksvensson commented on docs/input/blog/posts/_ViewStart.cshtml:
```diff
@@ -1,3 +1,3 @@
 @{
-    Layout = @$"_layout.cshtml";
+    Layout = $@"_layout.cshtml";
```
Why has this change been made?

FrankRay78 replied:
I could not get the documentation to build on my local machine, 

![image](https://github.com/user-attachments/assets/a4da1c50-0625-4bfb-84e8-cf1fc4970d51)

In desperation I then came across this, which looks like a dotnet/Razor breaking change: https://github.com/dotnet/razor/issues/10568

I'm on latest SDKs and Visual studio. 

Does the documentation still build for you?

```
C:\Users\info>dotnet --list-sdks
6.0.310 [C:\Program Files\dotnet\sdk]
6.0.407 [C:\Program Files\dotnet\sdk]
7.0.100 [C:\Program Files\dotnet\sdk]
8.0.400 [C:\Program Files\dotnet\sdk]

C:\Users\info>dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 7.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

![image](https://github.com/user-attachments/assets/84ffb884-c1f0-4642-a790-5b9c7e32e0e6)

FrankRay78 replied:
Razor syntax regression bug has now been addressed in the following commit to spectre.console main branch: 96512f353fd5a9e434ee6031fd8332c83ed4ec1e  

Hence why I'm marking this conversation as resolved.

FrankRay78 commented:
**IMPORTANT:** The following PR https://github.com/spectreconsole/examples/pull/1 should be reviewed/merged at the same time as this one.

FrankRay78 commented:
@patriksvensson I've addressed your earlier review comments and rebased against main, fixing conflicts. This PR is now ready for another review +/ merging.

**IMPORTANT:** The following PR https://github.com/spectreconsole/examples/pull/1 should be reviewed/merged at the same time as this one.
